site/app/libraries/database/DatabaseQueries.php

```diff
@@ -6474,12 +6474,58 @@ public function getUsersNotFullyGraded(Gradeable $gradeable, $component_id = "-1
   */
 
     public function getCurrentQueue() {
-        $this->course_db->query("SELECT ROW_NUMBER() OVER(order by time_in ASC),* FROM queue where current_state IN ('waiting','being_helped') order by ROW_NUMBER");
+        $query = "
+        SELECT ROW_NUMBER()
+            OVER (order by time_in ASC),
+                queue.*,
+                CASE
+                    WHEN helper.user_preferred_firstname IS NULL THEN helper.user_firstname
+                    ELSE helper.user_preferred_firstname
+                END AS helper_firstname,
+                CASE
+                    WHEN helper.user_preferred_lastname IS NULL THEN helper.user_lastname
+                    ELSE helper.user_preferred_lastname
+                END AS helper_lastname,
+                helper.user_group AS helper_group
+            FROM queue
+            LEFT JOIN users helper on helper.user_id = queue.help_started_by
+            WHERE current_state IN ('waiting','being_helped')
+            ORDER BY ROW_NUMBER
+        ";
+        $this->course_db->query($query);
         return $this->course_db->rows();
     }
 
     public function getPastQueue() {
-        $this->course_db->query("SELECT ROW_NUMBER() OVER(order by time_out DESC, time_in DESC),* FROM queue where time_in > ? AND current_state IN ('done') order by ROW_NUMBER", [$this->core->getDateTimeNow()->format('Y-m-d 00:00:00O')]);
+        $query = "
+        SELECT Row_number()
+            OVER (ORDER BY time_out DESC, time_in DESC),
+                queue.*,
+                CASE
+                    WHEN helper.user_preferred_firstname IS NULL THEN helper.user_firstname
+                    ELSE helper.user_preferred_firstname
+                END AS helper_firstname,
```

Comment:
The `COALESCE()` command may be more efficient here.  The same comment applies to the other queries being modified.